### PR TITLE
fix:  add replicate check permission

### PR DESCRIPTION
**Pull request by flywukong**
### Description

1. add permission check to replicate the data on secondary sp
2. add the err catch of HTTP write error of recovery or migration

### Rationale

fix permission

### Example

NA

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: modular/gater/admin_handler.go [hunk 1]
```diff
@@ -488,6 +487,46 @@ func (g *GateModular) replicateHandler(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		return ErrConsensusWithDetail("getSPID error: " + err.Error())
+	}
+
+	expectSecondarySPID := gvg.GetSecondarySpIds()[int(receiveTask.GetRedundancyIdx())]
+	if expectSecondarySPID != spID {
+		log.CtxErrorw(ctx, "secondary sp mismatch", "expect", expectSecondarySPID, "current", spID)
+		return ErrSecondaryMismatch
+	}
+
```
Comment (joeylichang) [inline modular/gater/admin_handler.go:526]: should add the check that the object is creted state, the object state should get from chain or metadata.

**Review by joeylichang — APPROVED** [review]

File: modular/gater/admin_handler.go [hunk 2]
```diff
@@ -488,6 +492,62 @@ func (g *GateModular) replicateHandler(w http.ResponseWriter, r *http.Request) {
 	log.CtxDebug(reqCtx.Context(), "succeed to replicate piece")
 }
 
+func (g *GateModular) checkReplicatePermission(ctx context.Context, receiveTask gfsptask.GfSpReceivePieceTask, signatureAddr string) error {
+	objectInfo, err := g.baseApp.Consensus().QueryObjectInfo(ctx, receiveTask.ObjectInfo.BucketName, receiveTask.ObjectInfo.ObjectName)
```
Comment (alexgao001) [inline modular/gater/admin_handler.go:496]: can use GfSpClient().GetObjectByID(objectId) and GfSpClient().GetGlobalVirtualGroup(bucketId, lvgId) to get GVG.

**Review by annielz — APPROVED** [review]

**Review by sysvm — APPROVED** [review]
LGTM

File: modular/gater/admin_handler.go [hunk 3]
```diff
@@ -488,6 +492,62 @@ func (g *GateModular) replicateHandler(w http.ResponseWriter, r *http.Request) {
...
+
+	if !receiveTask.BucketMigration {
+		// if it is replicate when uploading, the status should be created
+		if objectInfo.ObjectStatus != storagetypes.OBJECT_STATUS_CREATED {
+			return ErrNotCreatedState
+		}
+	} else {
+		// if it is bucket migration, the status should be sealed
+		if objectInfo.ObjectStatus != storagetypes.OBJECT_STATUS_SEALED {
+			return ErrNotCreatedState
```
Comment (will-2012) [inline modular/gater/admin_handler.go:510]: ErrNotSealedState

**Pull request merged**